src/frequenz/sdk/timeseries/_base_types.py

```diff
@@ -150,3 +152,37 @@ class Bounds(Generic[_T]):
 
     upper: _T
     """Upper bound."""
+
+
+@dataclass(frozen=True)
+class SystemBounds:
+    """Internal representation of system bounds for groups of components."""
+
+    # compare = False tells the dataclass to not use name for comparison methods
+    timestamp: datetime = dataclasses.field(compare=False)
+    """Timestamp of the metrics."""
+
+    inclusion_bounds: Bounds[Power] | None
+    """Total inclusion power bounds for all components of a pool.
+
+    This is the range within which power requests would be allowed by the pool.
+
+    When exclusion bounds are present, they will exclude a subset of the inclusion
+    bounds.
+    """
+
+    exclusion_bounds: Bounds[Power] | None
+    """Total exclusion power bounds for all components of a pool.
+
+    This is the range within which power requests are NOT allowed by the pool.
+    If present, they will be a subset of the inclusion bounds.
+    """
+
+
+class PoolType(enum.Enum):
+    """Enumeration of component pool types."""
+
+    BATTERY_POOL = "BATTERY_POOL"
+    EV_CHARGER_POOL = "EV_CHARGER_POOL"
+    PV_POOL = "PV_POOL"
+    CHP_POOL = "CHP_POOL"
```

Comment:
Is there any way to make this using a class instead where we can use different subclasses to customize the behavior for different types? Usually using enums makes the code harder to read in maintain, as it gets littered with `if`s (or now `match`es probably), or is this another case where the type will need to be sent over the wire?

Sorry, I'm not very familiar with this code to evaluate it myself :-/

But even if the type is sent over the wire, we could do the implementation using these "trait" classes, so we just have to map enum -> class now, but then the code still is clean and without a lot of `if`s for having different logic for different types.

Reply:
It is for accessing the instance of the pool, in here: https://github.com/frequenz-floss/frequenz-sdk-python/blob/aebdd00b19f6b8fe38eca3a2604e04923f14cf70/src/frequenz/sdk/actor/_power_managing/_power_managing_actor.py#L133-L137

It would be an if statement at startup, but I think any alternative would be a lot more work,  and I think this is intuitive, and that `_add_bounds_tracker` function would be the only place such an if-statement is necessary, so it is quite isolated as well.

And there won't be edge cases because PowerManager instances are created internally,  so we can control and test how and where it is initialized.